mgmt/share_config.c

```diff
@@ -165,8 +165,14 @@ static struct ksmbd_share_config *share_config_request(struct unicode_map *um,
 
 		share->path = kstrndup(ksmbd_share_config_path(resp), path_len,
 				      GFP_KERNEL);
-		if (share->path)
+		if (share->path) {
 			share->path_sz = strlen(share->path);
+			if (share->path_sz > 1 && share->path[share->path_sz - 1] == '/')
```

Comment:
@bonifaido Thanks for your patch:) BTW, if share pathname is "/home/me/Share//" or "/home/me/Share////", there no problem with this patch ?

Reply:
Sorry for the late reply.

Most probably there is a problem in those cases, is there a function available to do this in the module or in the kernel?

Reply:
there is no function for this. You need to add simple while loop like the following one. I didn't test it. can you check it ?

while (share->path_sz > 1 && share->path[share->path_sz - 1] == '/');
                   share->path_sz--;

Reply:
And please make the patch using git command to add signoff-by tag and your real name and mail address.
and need to add prefix(ksmbd: ) and patch description to the patch.
See other patches.
  https://github.com/namjaejeon/ksmbd/commit/ef5e1700ff3cdf9a401bc7d42c6c4db0f333a514

Reply:
Doing it right now!

Reply:
Thanks for your test:) I have optimized the patch and apply it to next branch.(https://github.com/namjaejeon/ksmbd/commit/3172144f392efcc33a4f768c22bd5efd4f5ca1e9)
If there is a problem, Let me know. Thanks.

Reply:
Thank you very much! What is the target kernel release for this patch?

Reply:
I will add stable tag to the patch. This patch will be applied to stable kernels(5.15, 6.1, 6.6, 6.9) and 6.10-rc1.